src/types.rs

```diff
@@ -258,73 +260,122 @@ fn map_address_from_binary(v: &Option<FixedSizeData<20>>, should_checksum: bool)
     })
 }
 
-fn map_binary<T: Hex>(v: &Option<T>) -> Option<String> {
+fn map_hex_string<T: Hex>(v: &Option<T>) -> Option<String> {
     v.as_ref().map(|v| v.encode_hex())
 }
 
+pub(crate) fn map_opt_result<T, V, F>(v: Option<T>, f: F) -> Result<Option<V>>
+where
+    F: Fn(T) -> Result<V>,
+{
+    match v {
+        Some(v) => Ok(Some(f(v)?)),
+        None => Ok(None),
+    }
+}
+
+fn map_i64_from_u64(v: Option<u64>) -> Result<Option<i64>> {
+    map_opt_result(v, |v| i64::try_from(v).context("converting u64 to i64"))
+}
+
+fn map_i64<T: AsRef<[u8]>>(v: &Option<T>) -> Result<Option<i64>> {
+    map_opt_result(v.as_ref(), |v| {
+        v.as_ref()
+            .try_into()
+            .context("converting to fixed size bytes i64")
+            .map(i64::from_be_bytes)
+    })
+}
+
+fn map_bigint_u64<T: AsRef<[u8]>>(v: &Option<T>) -> Result<Option<BigInt>> {
+    map_opt_result(v.as_ref(), |v| {
+        v.as_ref()
+            .try_into()
+            .context("converting to fixed size bytes u64")
+            .map(u64::from_be_bytes)
+            .map(BigInt::from)
+    })
+}
+
+fn map_bigint_u128<T: AsRef<[u8]>>(v: &Option<T>) -> Result<Option<BigInt>> {
+    map_opt_result(v.as_ref(), |v| {
+        v.as_ref()
+            .try_into()
+            .context("converting to fixed size bytes u128")
+            .map(u128::from_be_bytes)
+            .map(BigInt::from)
+    })
+}
```

Comment:
@ozgrakkurt my only question is whether there is any value in mapping to u64 then to BigInt vs u128 then BigInt?

Reply:
And I would also love it if you could double check the places where I used these two conversions and let me know if you think there is any possibility of overflow. I believe there's enough room though.

Reply:
Ideally want to first use `ruint::aliases::U256::from_be_slice` then use `convert_bigint_signed` and `convert_bigint_unsigned` functions that are already in the library code. Don't need to use u64 or u128